.github/workflows/ruff.yml

```diff
@@ -0,0 +1,15 @@
+name: ruff
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: pip install --user ruff
+    - run: ruff --format=github .
```

Comment:
Do we need this at all given that it's run in `.github/workflows/codespell-private.yml` already?

Reply:
I do not really understand the purpose / differentiation of "private" but it is only testing `--select=ANN` which is a much narrower set of rules than this Ruff action.

I disabled ANN101 in one and ANN401 in the other so we could see that both do their thing.